clippy.toml

```diff
@@ -1,4 +1,32 @@
 doc-valid-idents = [
   "PyPI",
   ".." # Include the defaults
-] 
+]
+
+disallowed-types = [
+  "std::fs::DirEntry",
+  "std::fs::File",
+  "std::fs::OpenOptions",
+  "std::fs::ReadDir",
+]
```

Comment:
What's the motivation for disallowing these types? Are there equivalent types in `fs_err`? I see that you explicitly allow them further down.

Reply:
Yes, there are equivalent types (https://docs.rs/fs-err/2.11.0/fs_err/#structs), but they do not seem to be fully compatible with types in `std::fs` (I got errors when replacing them). So I thought it would be good to use the types in `fs_err` as much as possible and explicitly allow types in `std::fs` otherwise.
But I think it might be a bit annoying to have to explicitly allow it every time we use types in `std::fs`. If so, I'll revise it.

Reply:
Let's hear what @konstin thinks. He knows best when and where we should use `fs_err`.

Reply:
`std::fs::File::create` doesn't tell you the filename when an error occurs, while `fs::File::create` does. There are case where generics are implemented on std types (e.g. `AutoStream`) so we need to "downcast". I like the current solution.

Reply:
The clippy feature we're missing to ban `std::fs` as a module is https://github.com/rust-lang/rust-clippy/issues/9489